src/App.tsx

```diff
@@ -15,48 +16,109 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphabetically = 'Alphabetically',
+  ByLength = 'ByLength',
+  Default = '',
+}
+interface SortState {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: SortState,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Alphabetically:
+          return good1.localeCompare(good2);
+
+        case SortType.ByLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const preparedGoods = getPreparedGoods(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortField !== SortType.Alphabetically },
+          )}
+          onClick={() => setSortField(SortType.Alphabetically)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn(
+            'button is-success',
+            { 'is-light': sortField !== SortType.ByLength },
+          )}
+          onClick={() => setSortField(SortType.ByLength)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn(
+            'button is-warning',
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
I think it could be the same function as for handleSort
```
  const handleSort = (type: typeof SortType[keyof typeof SortType]) => {
    if (type === SortType.SORT_REVERSE) {
      setIsReversed(!isReversed);
    } else {
      setSortField(type);
    }
  };
```